server.go

```diff
@@ -222,8 +222,14 @@ func (s *Server) Close() error {
 		return ErrServerClosed
 	}
 
+	// Stop goroutines.
 	if s.rpDone != nil {
 		close(s.rpDone)
+		s.rpDone = nil
+	}
+	if s.done != nil {
+		close(s.done)
+		s.done = nil
 	}
```

Comment:
The `s.done` channel gets closed by the `setClient()`. I think it was moved there originally because we potentially changed the client depending on "local mode" or "cluster mode". That doesn't happen anymore so we should move the channel creation to `Open()` and the channel close here to `Close()`.